### PR TITLE
500 while waiting for all content to be ready

### PR DESCRIPTION
**Pull request by wheresrhys**
@adgad @mattc a good idea to avoid the following? Fastly *should* cover our backs

image

### CODE REVIEW

File: server/controllers/uk-front.js [hunk 1]
```diff
@@ -32,6 +32,10 @@ var topics = [
 
 module.exports = function(req, res) {
 	Metrics.instrument(res, { as: 'express.http.res' });
+	if (!bigRead.items || !lex.items || !management.items || !frontPageSkyline.items || !topStories.items || !comment.items || !lunch.items || !personInNews.items) {
```
Comment (commuterjoy) [inline server/controllers/uk-front.js]: how confident are we of the stale-if-error handling? I've only really got anecdotal evidence it's working.

**Comment by commuterjoy** [comment]
Really not sure about this.

A better short term option would be to turn on preboot for engels - https://devcenter.heroku.com/articles/preboot 

I don't really understand why that top stories query fails so much, so frequently.

**Comment by matthew-andrews** [comment]
does moving to elastic search should have solved this for us?

**Comment by wheresrhys** [comment]
I'll elasticate it all properly today. Would preboot help though? It seems
more geared towards downtime than bad behaviour once up

I could put this strict conditional behind a flag, we can work on how to
properly test stale while re validate, then turn it on for a bit to see? Or
even better (for testing purposes) put res.send(500) behind a flag. Worth
doing before public alpha

On Thursday, February 12, 2015, Matt Andrews <notifications@github.com>
wrote:

> does moving to elastic search should have solved this for us?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Financial-Times/next-engels/pull/156#issuecomment-74126510>
> .
>

-- 

------------------------------
This email was sent by a company owned by Pearson plc, registered office at 
80 Strand, London WC2R 0RL.  Registered in England and Wales with company 
number 53723.

**Pull request closed without merging**